src/datasets/arrow_dataset.py

```diff
@@ -3553,6 +3555,46 @@ def delete_file(file):
             )
         return repo_id, split, uploaded_size, dataset_nbytes
 
+    def get_remote_dataset_infos(
```

Comment:
I've hesitated to add this to `DatasetInfo.from_hub()`. The only thing I didn't like about it was that the return value is nullable. I can still make the change if you think it makes more sense.